src/pyqasm/visitor.py

```diff
@@ -352,6 +366,17 @@ def _visit_quantum_register(
 
         self._module._add_qubit_register(register_name, register_size)
 
+        # Inline: Update offsets after adding a new register if device_qubits is set
+        if self._consolidate_qubits:
+            self._qubit_register_offsets[register_name] = self._qubit_register_max_offset
+            self._qubit_register_max_offset += register_size
+            # offsets = {}
```

Comment:
You can remove the old code and just add a explanation as to why `self._qubit_register_offsets` is needed